exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -307,3 +315,235 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
+			return nil, errors.Wrapf(err, "failed to create new Machine for infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+		}
+
+		machines = append(machines, *machine)
+	}
+
+	return machines, nil
+}
+
+// ensureInfraMachineOnwerRefs sets the ownerReferences on the each infraMachine to its associated MachinePool Machine if it hasn't already been done.
+func (r *MachinePoolReconciler) ensureInfraMachineOnwerRefs(ctx context.Context, updatedMachines []clusterv1.Machine, infraMachines []unstructured.Unstructured) error {
+	log := ctrl.LoggerFrom(ctx)
+	infraMachineNameToMachine := make(map[string]clusterv1.Machine)
+	for _, machine := range updatedMachines {
+		infraRef := machine.Spec.InfrastructureRef
+		infraMachineNameToMachine[infraRef.Name] = machine
+	}
+
+	for i := range infraMachines {
+		infraMachine := &infraMachines[i]
+		ownerRefs := infraMachine.GetOwnerReferences()
+
+		machine, ok := infraMachineNameToMachine[infraMachine.GetName()]
+		if !ok {
+			return errors.Errorf("failed to patch ownerRef for infraMachine %q because no Machine has an infraRef pointing to it", infraMachine.GetName())
+		}
+		machineRef := metav1.NewControllerRef(&machine, machine.GroupVersionKind())
+		if !util.HasOwnerRef(ownerRefs, *machineRef) {
+			log.V(2).Info("Setting ownerRef on infraMachine", "infraMachine", infraMachine.GetName(), "namespace", infraMachine.GetNamespace(), "machine", machine.GetName())
+			ownerRefs = util.EnsureOwnerRef(ownerRefs, *machineRef)
+			infraMachine.SetOwnerReferences(ownerRefs)
+			if err := r.Client.Update(ctx, infraMachine); err != nil {
+				return errors.Wrapf(err, "failed to update infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+			}
+		}
+	}
+
+	return nil
+}
+
+// getNewMachine creates a new Machine object. The name of the newly created resource is going
+// to be created by the API server, we set the generateName field.
+func getNewMachine(mp *expv1.MachinePool, infraMachine *unstructured.Unstructured) *clusterv1.Machine {
+	infraRef := corev1.ObjectReference{}
+	if infraMachine != nil {
+		infraRef.APIVersion = infraMachine.GetAPIVersion()
+		infraRef.Kind = infraMachine.GetKind()
+		infraRef.Name = infraMachine.GetName()
+		infraRef.Namespace = infraMachine.GetNamespace()
+	}
+
+	annotations := mp.Spec.Template.Annotations
+
+	machine := &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", mp.Name),
+			// Note: by setting the ownerRef on creation we signal to the Machine controller that this is not a stand-alone Machine.
+			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(mp, mp.GroupVersionKind())},
+			Namespace:       mp.Namespace,
+			Labels:          make(map[string]string),
+			Annotations:     annotations,
+		},
+		Spec: clusterv1.MachineSpec{
+			ClusterName:       mp.Spec.ClusterName,
+			InfrastructureRef: infraRef,
+		},
+	}
+
```

Comment:
I'm not sure I understand, where would the annotation be added and how would it get back to the MachinePool in the example you mentioned?

Reply:
With your current code if something adds the following:
```go
machine.Annotations["annotationkey"] = "annotationValue"
```

This would also modify the annotations in the MachinePool as you currently share the exact same map between Machine.annotations and MachinePool.Spec.Template.Annotations